### PR TITLE
[Core] Adding 64 bit version of Hash FNV1A

### PR DESCRIPTION
**Pull request by loumalouomega**
Prime and hash from https://gist.github.com/underscorediscovery/81308642d0325fd386237cfa3b44785c

As mentioned in #1812

### CODE REVIEW

File: kratos/tests/containers/test_data_value_container.cpp [hunk 1]
```diff
@@ -38,5 +42,50 @@ KRATOS_TEST_CASE_IN_SUITE(DataValueContainer, KratosCoreFastSuite) {
   for (std::size_t i = 0; i < distances.size(); i++)
     KRATOS_CHECK_EQUAL(distances[i], original_distances[i]);
 }
+
+/**
+ * This test checks recursively the proper functioning of the DataValueContainer
+ */
+KRATOS_TEST_CASE_IN_SUITE(DataValueContainerRecursive, KratosCoreFastSuite) {
```
Comment (pooyan-dadvand) [inline kratos/tests/containers/test_data_value_container.cpp]: why recursive? I don't see any recursion here....

File: kratos/tests/containers/test_data_value_container.cpp [hunk 2]
```diff
@@ -38,5 +42,50 @@ KRATOS_TEST_CASE_IN_SUITE(DataValueContainer, KratosCoreFastSuite) {
...
+  DataValueContainer container;
+  Vector original_distances(4);
+  original_distances[0] = 0.00;
+  original_distances[1] = 0.10;
+  original_distances[2] = 0.20;
+  original_distances[3] = 0.30;
+
+  const std::size_t number_recursions = 1000;
```
Comment (pooyan-dadvand) [inline kratos/tests/containers/test_data_value_container.cpp]: Why you should repeat this 1000 times?

**Review by pooyan-dadvand — COMMENTED** [review]
Some minor comments in tests

File: kratos/tests/containers/test_data_value_container.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/tests/containers/test_data_value_container.cpp]: For the loop, I was trying to check that the origin of the error wasn't when overwrite several times the same value on memory

File: kratos/tests/containers/test_data_value_container.cpp (hunk 2 above)
Comment (loumalouomega) [reply kratos/tests/containers/test_data_value_container.cpp]: The number is just arbitrary large, I was trying to check that the memory is not corrupted doing the same several times

File: kratos/tests/containers/test_data_value_container.cpp (hunk 1 above)
Comment (pooyan-dadvand) [reply kratos/tests/containers/test_data_value_container.cpp]: Repeated may be better, recursive reflects to me as calling the function inside itself...

Meanwhile I don't think these stress tests are useful at all....

File: kratos/tests/containers/test_data_value_container.cpp (hunk 2 above)
Comment (pooyan-dadvand) [reply kratos/tests/containers/test_data_value_container.cpp]: You will see a memory collapse by valgrind in few iterations (say 1 to 3) and usually such a stress tests are for finding concurrency problems.

Comment (loumalouomega) [reply kratos/tests/containers/test_data_value_container.cpp]: OK, reverting

**Review by pooyan-dadvand — APPROVED** [review]
Thanks for adding the 64bit version. Just to add that this should not replace the one used in variable even for 64 bit compilation!

**Comment by loumalouomega** [comment]
Of course not, different classes, different names

**Pull request merged**